src/main/java/com/google/sps/servlets/EditProfileServlet.java

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.User;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/EditProfile")
+public class EditProfileServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+    PrintWriter out = response.getWriter();
+    out.println("<h1>Set Nickname</h1>");
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      
+      User user = getUserInfo(userService.getCurrentUser().getUserId());
+      out.println("<p>Set your nickname here:</p>");
+      out.println("<form method=\"POST\" action=\"/EditProfile\">");
+      out.println("<input name=\"nickname\" value=\"" + user.getName() + "\" />");
+      out.println("<br/>");
+      out.println("<button>Submit</button>");
+      out.println("</form>");
```

Comment:
Since we now use server-side rendered HTML, we should integrate this login functionality with the overall page and servlets. For example in the `WelcomeServlet` I've created a dummy `User` object.

```
		LandingPageModel model = new LandingPageModel(new User("Patrick", 42L));
```

Update the `User` with a new attribute `loggedIn`, and set it based on the `userService` result. When the user *is* logged in, also populate the user name (email?) and numeric ID -- assuming they are different.
If the user is not logged in, also put the LoginUrl into the user object.

In `templates/common/header.ftl`  add the user "widget" to the toolbar:
 - If the user is not logged in, have the template output the login link
 - if the user is logged in, show their name instead.
You can use the FreeMarker template syntax (<#if ...>) to switch between the scenarios.

We most likely won't need the EditProfileService right away, until we have some settings for the user to customize.
(When we do get to that -- or if you want to do it right away -- it should use another FreeMarker template ("profile.ftl") with all of the HTML, and the servlet should use the html template renderer to produce html.)